internal/service/mongodb/mongodb.go

```diff
@@ -1019,9 +1046,52 @@ func (m *mongodbResourceModel) refreshFromOutput(ctx context.Context, output *vm
 			Uptime:          types.StringPointerValue(server.Uptime),
 			CreateDate:      types.StringPointerValue(server.CreateDate),
 		}
+
+		if server.CloudMongoDbServerRole != nil {
+			mongoServerInstance.ServerRole = types.StringPointerValue(server.CloudMongoDbServerRole.CodeName)
+			if *server.CloudMongoDbServerRole.Code == "A" || *server.CloudMongoDbServerRole.Code == "MB" {
+				m.MemberProductCode = types.StringPointerValue(server.CloudMongoDbProductCode)
+				memberCount++
+			} else if *server.CloudMongoDbServerRole.Code == "AB" {
+				m.ArbiterProductCode = types.StringPointerValue(server.CloudMongoDbProductCode)
+				arbiterCount++
+			} else if *server.CloudMongoDbServerRole.Code == "RT" {
+				m.MongosProductCode = types.StringPointerValue(server.CloudMongoDbProductCode)
+				mongosCount++
+			} else if *server.CloudMongoDbServerRole.Code == "C" {
+				m.ConfigProductCode = types.StringPointerValue(server.CloudMongoDbProductCode)
+				configCount++
+			}
+		}
+		if server.ClusterRole != nil {
+			mongoServerInstance.ClusterRole = types.StringPointerValue(server.ClusterRole.Code)
+		}
 		serverList = append(serverList, mongoServerInstance)
 	}
 
+	if output.ClusterType != nil {
+		m.ClusterTypeCode = types.StringPointerValue(output.ClusterType.Code)
+		if *output.ClusterType.Code == "SHARDED_CLUSTER" {
+			memberCount = memberCount / int64(*output.ShardCount)
+			if arbiterCount > 0 {
+				arbiterCount = arbiterCount / int64(*output.ShardCount)
+			}
+		}
+	}
+	m.MemberServerCount = types.Int64Value(memberCount)
+	m.ArbiterServerCount = types.Int64Value(arbiterCount)
+	m.MongosServerCount = types.Int64Value(mongosCount)
+	m.ConfigServerCount = types.Int64Value(configCount)
+	if arbiterCount == 0 {
+		m.ArbiterProductCode = types.StringValue("not allocated")
+	}
+	if mongosCount == 0 {
+		m.MongosProductCode = types.StringValue("not allocated")
+	}
+	if configCount == 0 {
+		m.ConfigProductCode = types.StringValue("not allocated")
+	}
```

Comment:
This part would be better to be factored out as a function like `Int64ZeroFromInt32` and please add some comments why this value should be used instead of zero value.